irc/metadata.go

```diff
@@ -0,0 +1,257 @@
+package irc
+
+import (
+	"errors"
+	"regexp"
+	"slices"
+	"strings"
+
+	"maps"
+
+	"github.com/ergochat/ergo/irc/caps"
+	"github.com/ergochat/ergo/irc/modes"
+	"github.com/ergochat/ergo/irc/utils"
+)
+
+var (
+	errMetadataTooManySubs = errors.New("too many subscriptions")
+)
+
+type MetadataStore = map[string]string
+
+type MetadataThing = struct {
```

Comment:
This is dead code now....I don't think we're going to implement visibility as part of the MVP for this so we can just use `map[string]string` for now